sync/evmdownloader.go

```diff
@@ -243,59 +247,74 @@ func NewEVMDownloaderImplementation(
 	blockFinality *big.Int,
 	waitForNewBlocksPeriod time.Duration,
 	appender LogAppenderMap,
-	topicsToQuery []common.Hash,
-	adressessToQuery []common.Address,
+	addressesToQuery []common.Address,
 	rh *RetryHandler,
+	finalizedBlockType *big.Int,
 ) *EVMDownloaderImplementation {
 	logger := log.WithFields("syncer", syncerID)
+	var topics []common.Hash
+	if appender != nil {
+		topics = appender.GetTopics()
+	}
+
 	return &EVMDownloaderImplementation{
 		ethClient:              ethClient,
 		blockFinality:          blockFinality,
 		waitForNewBlocksPeriod: waitForNewBlocksPeriod,
 		appender:               appender,
-		topicsToQuery:          topicsToQuery,
-		adressessToQuery:       adressessToQuery,
+		topicsToQuery:          topics,
+		addressesToQuery:       addressesToQuery,
 		rh:                     rh,
 		log:                    logger,
+		finalizedBlockType:     finalizedBlockType,
 	}
 }
 
 func (d *EVMDownloaderImplementation) ChainID(ctx context.Context) (uint64, error) {
 	chainID, err := d.ethClient.ChainID(ctx)
-	if err != nil || chainID == nil {
-		return 0, fmt.Errorf("fail to get chainID from ethClient. Err: %w", err)
+	if err != nil {
+		return 0, fmt.Errorf("failed to retrieve chain id. Err: %w", err)
+	}
+
+	if chainID == nil {
+		return 0, errors.New("chain id is undefined")
```

Comment:
Error to a global var?

Reply:
Fixed https://github.com/agglayer/aggkit/pull/357/commits/aa160bd90b79c2074252212341cfe547d2598eb3